src/bun.js/ipc.zig

```diff
@@ -194,21 +194,29 @@ const json = struct {
     // 1 is regular
     // 2 is internal
 
-    pub fn decodeIPCMessage(
-        data: []const u8,
-        globalThis: *JSC.JSGlobalObject,
-    ) IPCDecodeError!DecodeIPCMessageResult {
+    pub fn decodeIPCMessage(data: []const u8, globalThis: *JSC.JSGlobalObject) IPCDecodeError!DecodeIPCMessageResult {
         if (bun.strings.indexOfChar(data, '\n')) |idx| {
             var kind = data[0];
-            var json_data = data[1..idx];
+            var json_data = data[0..idx];
 
             switch (kind) {
-                1, 2 => {},
+                2 => {
+                    json_data = data[1..idx];
+                },
+                '{', //object
+                '[', //array
+                '"', //string
+                '0'...'9', //number
+                't', //true
+                'f', //false
+                'n', //null
```

Comment:
What is the purpose of checking this? Can we treat all values that are not 2 as kind 1? There are other valid first bytes to JSON, such as `-` and since the branches are no longer any different, we don't need to do this work here anymore